### PR TITLE
Scaling the Lienzo

### PR DESCRIPTION
**Pull request by nor0x**
Hi @ylatuya - not sure if you are open for PRs - I just wanted to contribute my work to this great project. I created a fork where I would implement basic transformations like scaling, translation and rotation to the Lienzo itself. I just finished the scaling operation which I have tested on iOS and macOS.

Thanks for your great work.

### CODE REVIEW

**Comment by ylatuya** [comment]
Hi, I think the API is mostly stable now and I am open for PR's :)
I still have to fill the CONTRIBUTING.md for contributions and create a proper .editorconfig to standarize code formatting across contributors.
So far I like the idea of this PR! Let me find some time to review it this week. What is 100% needed before merge are functional tests for the new feature :)

**Comment by nor0x** [comment]
Thanks for getting back. I will wait for your review and may do some functional tests in the meantime

**Comment by ylatuya** [comment]
> Thanks for getting back. I will wait for your review and may do some functional tests in the meantime

LienzoViewIntegrationScenario would be the perfect place to test this functionality and LienzoSharp.Tests.Utils would be the right place to implement The_user_pinches () that will trigger the events in the mock.

File: src/LienzoSharp.Core/Abstractions/ICanvas.cs [hunk 1]
```diff
@@ -125,6 +125,16 @@ public interface ICanvas : IDisposable
...
         void Scale(double x, double y);
 
+        /// <summary>
+        /// Scales the canvas with the specified x and y at the specified x and y coordinates.
+        /// </summary>
+        /// <param name="x">The x coordinate.</param>
+        /// <param name="y">The y coordinate.</param>
+        /// <param name="originX">Origin x.</param>
+        /// <param name="originY">Origin y.</param>
+        void Scale(double x, double y, double originX, double originY);
```
[the remark this answers is not in the document]
Comment (ylatuya) [reply src/LienzoSharp.Core/Abstractions/ICanvas.cs]: After reviewing everything, I think this isn't needed in the ICanvas api, but instead in the IMatrix one

[the remark this answers is not in the document]
Comment (nor0x) [reply src/LienzoSharp.Core/Abstractions/ICanvas.cs]: That's cool - it's great to see projects like these being developed in the open and I'm happy to contribute. And based on your comments I can learn a lot of things and best practices . I will look into the reviews tomorrow.

**Comment by ylatuya** [comment]
I was also thinking that, since scaling is not yet implemented in the RectangleTransformer, it would be easier to support translation or rotation than scalling :)

File: src/LienzoSharp.Core/LienzoCanvas.cs [hunk 2]
```diff
@@ -152,6 +167,42 @@ void BindView(IView newView)
             }
         }
 
+        void HandleScaling(object sender, ScaleArgs e)
+        {
+            if (e.Status == GestureStatus.Started)
+            {
+                startScale = lienzoView.Model.CTM.ScaleX;
```
[the remark this answers is not in the document]
Comment (ylatuya) [reply src/LienzoSharp.Core/LienzoCanvas.cs]: I was looking on how the Pinch event emits properties an I don't think we can use the same Transformers approach. This event only emits Scale and ScaleOrigin so it's not possible to actually emulate some one dragging 2 anchors with its fingers as I wanted to do.
So I would continue with your current approach, trying to calculate the correct scale and translation values without the need of OriginX and OriginY :)

**Comment by ylatuya** [comment]
This documentation of SkiaSharp is also very good to understand how the transformation works and how you can express any affine transformation in a 2D space with only Scale, Transformation and Skew https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/graphics/skiasharp/transforms/matrix

**Pull request closed without merging**